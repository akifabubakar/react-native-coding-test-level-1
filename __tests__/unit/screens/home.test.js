import { render, fireEvent, screen } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import HomeScreen from "../../../src/screens/home";
import AppNavigator from "../../../src/navigation";

describe("<HomeScreen/>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("onPress should navigate to Contact Us Screen", async () => {
    const component = <AppNavigator />;

    render(component);
    const toClick = await screen.findByText("Contact Us");

    fireEvent(toClick, "press");
    const newHeader = await screen.getByText("Contact Us");

    expect(newHeader).toBeTruthy();
  });
});
