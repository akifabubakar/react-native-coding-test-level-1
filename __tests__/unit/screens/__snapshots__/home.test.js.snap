// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "aliceblue",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "borderRadius": 20,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 18,
      }
    }
    testID="navigateButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Contact Us
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`<HomeScreen/> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "aliceblue",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "borderRadius": 20,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 18,
      }
    }
    testID="navigateButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Contact Us
    </Text>
  </View>
</RCTSafeAreaView>
`;
